tasks/installation.yml

```diff
@@ -1,33 +1,86 @@
 ---
 # tasks file for sap-preconfigure: installation
 
-- name: ensure minimal required packages are installed
+- name: Ensure required package groups are installed
   package:
     state: present
-    name: "{{ sap_preconfigure_min_pkgs }}"
-  when:
-    - sap_preconfigure_min_package_check|bool
-    - not( (sap_preconfigure_min_pkgs is undefined) or (sap_preconfigure_min_pkgs is none) or (sap_preconfigure_min_pkgs | trim == '') )
-
-- name: ensure required package groups are installed
-  package:
-    state: present
-    name: "{{ sap_preconfigure_packagegroups }}"
-  check_mode: "{{ sap_preconfigure_installation_check_only }}"
+    name: "{{ __sap_preconfigure_packagegroups }}"
 
-- name: ensure required packages are installed
+- name: Ensure required packages are installed
   package:
     state: present
-    name: "{{ sap_preconfigure_packages }}"
-  check_mode: "{{ sap_preconfigure_installation_check_only }}"
+    name: "{{ __sap_preconfigure_packages }}"
   register: sap_preconfigure_register_groupinstall
 
-- name: ensure system is updated to the latest patchlevel
+- name: Ensure minimum packages are installed
+  block:
+    - name: check if minimum release needs to be installed
+      shell: |
+             set -x
+             required_pkg={{ pkg | join('-') }}
+             newest=$(echo -e "$required_pkg\n$(rpm -q {{ pkg[0] }} )"| sort -V | tail -1)
+             if [ $newest == $required_pkg ]; then
+                echo $newest
+             fi
+      loop: "{{ __sap_preconfigure_min_pkgs }}"
+      loop_control:
+               loop_var: pkg
```

Comment:
indentation

Reply:
solved in https://github.com/berndfinger/sap-preconfigure/commit/ef59fa41c5d4144f509367fc1772fc78f6039f80 .